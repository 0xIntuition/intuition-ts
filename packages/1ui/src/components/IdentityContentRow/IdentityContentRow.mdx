import { Canvas, Controls, Meta, Source, Title } from '@storybook/blocks'

import { Button } from '..'

import * as Stories from './IdentityContentRow.stories'

<Meta of={Stories} />

<Title />

#### A short description goes here.

<Button
  variant="accent"
  onClick={() => window.open('link_goes_here', '_blank')}
>
  {'API Reference'}
</Button>

<Source
  dark
  language="tsx"
  code={`
import { IdentityContentRow } from '@0xintuition/1ui'

<IdentityContentRow />
`} />

<Canvas of={Stories.BasicUsage} sourceState="none" />

## Properties

<Controls />

## Variants

### Other

<Source
  dark
  language="tsx"
  code={`
<IdentityContentRow variant="other">
`}
/>

<Canvas of={Stories.Other} sourceState="none" />
