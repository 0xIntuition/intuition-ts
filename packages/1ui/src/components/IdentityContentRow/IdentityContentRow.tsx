import * as React from 'react'

import { cn } from 'styles'
import { CurrencyType, Identity, IdentityType } from 'types'

import {
  Avatar,
  Button,
  ButtonVariant,
  Copy,
  HoverCard,
  HoverCardContent,
  HoverCardTrigger,
  Icon,
  IdentityValueDisplay,
  ProfileCard,
  TagsContent,
  TagWithValue,
  TagWithValueProps,
  TextVariant,
  Trunctacular,
} from '..'

export interface IdentityContentRowProps
  extends React.HTMLAttributes<HTMLDivElement> {
  variant?: IdentityType
  amount: number
  currency?: CurrencyType
  name: string
  description: string
  id: string
  claimLink?: string
  avatarSrc: string
  link: string
  ipfsLink: string
  totalFollowers: number
  tags?: TagWithValueProps[]
}

interface NameAndAddressProps {
  name: string
  id: string
  link: string
  ipfsLink: string
  hasTags: boolean
}

const NameAndAddress = ({
  name,
  id,
  link,
  ipfsLink,
  hasTags,
}: NameAndAddressProps) => {
  return (
    <div
      className={cn(
        'mb-1 flex',
        hasTags ? 'flex-row items-center' : 'flex-col',
      )}
    >
      <a href={link}>
        <Trunctacular
          value={name}
          variant={TextVariant.bodyLarge}
          className="mr-2"
          maxStringLength={42}
        />
      </a>
      <div className="flex flex-row gap-1 items-center">
        <a href={ipfsLink} target="_blank" rel="noreferrer noreopener">
          <Trunctacular
            value={id}
            className="text-secondary-foreground"
            maxStringLength={16}
          />
        </a>
        <Copy text={id} />
      </div>
    </div>
  )
}

const IdentityContentRow = ({
  variant = Identity.user,
  amount,
  currency,
  name,
  description,
  id,
  claimLink,
  avatarSrc,
  link,
  ipfsLink,
  totalFollowers,
  tags,
  children,
  className,
  ...props
}: IdentityContentRowProps) => {
  const hasTags = !!(tags && tags.length > 0)

  const content = (
    <div
      className={cn(
        `w-full flex justify-between items-center max-sm:flex-col max-sm:gap-3`,
        className,
      )}
      {...props}
    >
      <div className="flex items-center">
        <HoverCard openDelay={100} closeDelay={100}>
          <HoverCardTrigger asChild>
            <a href={claimLink || link}>
              <Avatar
                variant={variant}
                src={avatarSrc}
                name={name}
                className="mr-4 w-[64px] h-[64px]"
              />
            </a>
          </HoverCardTrigger>
          <HoverCardContent side="right" className="w-max">
            <div className="flex flex-col gap-4 w-80 max-md:w-[80%]">
              <ProfileCard
                variant={variant}
                avatarSrc={avatarSrc ?? ''}
                name={name}
                id={id ?? ''}
                bio={description ?? ''}
                ipfsLink={ipfsLink}
                className="profile-card"
              />
              {link && (
                <a href={link}>
                  <Button variant={ButtonVariant.secondary} className="w-full">
                    View Identity{' '}
                    <Icon name={'arrow-up-right'} className="h-3 w-3" />
                  </Button>
                </a>
              )}
            </div>
          </HoverCardContent>
        </HoverCard>
        <div className="flex flex-col">
          <NameAndAddress
            name={name}
            id={id}
            link={claimLink || link}
            ipfsLink={ipfsLink}
            hasTags={hasTags}
          />
          {hasTags && (
            <div className="flex gap-2 mt-1">
              <TagsContent numberOfTags={tags.length} link={link}>
                {tags.slice(0, 2).map((tag, index) => (
                  <TagWithValue
                    label={tag.label}
                    value={tag.value}
                    key={index}
                  />
                ))}
              </TagsContent>
            </div>
          )}
        </div>
      </div>

      <IdentityValueDisplay
        value={amount}
        currency={currency}
        followers={totalFollowers}
      />
    </div>
  )

  return (
    <div className="w-full">
      {content}
      {children}
    </div>
  )
}

export { IdentityContentRow }
