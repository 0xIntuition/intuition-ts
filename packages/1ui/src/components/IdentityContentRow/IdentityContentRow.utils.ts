export const IdentityContentVariant = {
  entity: 'entity',
  user: 'user',
} as const
