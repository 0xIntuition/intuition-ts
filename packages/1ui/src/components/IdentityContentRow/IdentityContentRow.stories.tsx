import React from 'react'

import type { Meta, StoryObj } from '@storybook/react'
import { Identity } from 'types'

import { IdentityContentRow } from './IdentityContentRow'

const meta: Meta<typeof IdentityContentRow> = {
  title: 'Components/Identity/IdentityContentRow',
  component: IdentityContentRow,
  argTypes: {
    variant: {
      description: 'Variant of avatar',
      options: Object.values(Identity),
      table: {
        type: { summary: 'string' },
        defaultValue: { summary: 'user' },
      },
      control: 'select',
    },
  },
}

export default meta

type Story = StoryObj<typeof IdentityContentRow>

// Example story for the default state
export const UserVariant: Story = {
  args: {
    variant: 'user',
    name: 'John Doe',
    id: '0x1234567890abcdef1234567890abcdef12345678',
    avatarSrc: 'https://avatars.githubusercontent.com/u/94311139?s=200&v=4"',
    amount: 1.21,
    totalFollowers: 305,
    tags: [
      { label: 'keyboard', value: 34 },
      { label: 'ergonomic', value: 56 },
      { label: 'wireless', value: 12 },
      { label: 'gaming', value: 77 },
      { label: 'work', value: 11 },
      { label: 'home', value: 34 },
    ],
  },
  render: (args) => (
    <div className="w-[800px]">
      <IdentityContentRow {...args}></IdentityContentRow>
    </div>
  ),
}

export const EntityVariant: Story = {
  args: {
    variant: 'non-user',
    name: 'Amazon',
    id: '0x1234567890abcdef1234567890abcdef12345678',
    amount: 0.321,
    totalFollowers: 123,
    tags: [
      { label: 'keyboard', value: 34 },
      { label: 'ergonomic', value: 56 },
      { label: 'wireless', value: 12 },
      { label: 'gaming', value: 77 },
      { label: 'work', value: 11 },
      { label: 'home', value: 34 },
    ],
  },
  render: (args) => (
    <div className="w-[800px]">
      <IdentityContentRow {...args}></IdentityContentRow>
    </div>
  ),
}
