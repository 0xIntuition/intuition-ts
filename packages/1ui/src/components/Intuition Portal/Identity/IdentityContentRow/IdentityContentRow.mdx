import { Canvas, Controls, Meta, Source, Title } from '@storybook/blocks'

import { Button } from 'components'

import * as Stories from './IdentityContentRow.stories'

<Meta of={Stories} />

<Title />

#### An Identity Content Row is a component that displays the identity information for a user or an entity

<Source
  dark
  language="tsx"
  code={`
import { IdentityContentRow } from '@0xintuition/1ui'
import { Text } from 'components/Text'

<IdentityContentRow
  variant="user"
  avatarSrc="https://avatars.githubusercontent.com/u/94311139?s=200&v=4"
  name="John Doe"
  walletAddress="0x1234567890abcdef1234567890abcdef12345678"
  amount={1.21}
  totalFollowers={305}
  tags={[
    { label: 'keyboard', value: 34 },
    { label: 'ergonomic', value: 56 },
    { label: 'wireless', value: 12 },
    { label: 'gaming', value: 77 },
    { label: 'work', value: 11 },
    { label: 'home', value: 34 },
  ]}
>
  <Text variant="body" className="text-primary-foreground">
    Extra Content
  </Text>
</IdentityContentRow>
`} />

<Canvas of={Stories.UserVariant} sourceState="none" />

## Properties

<Controls />

## Variants

### Non User

<Source
  dark
  language="tsx"
  code={`
import { IdentityContentRow } from '@0xintuition/1ui'
import { Text } from 'components/Text'

<IdentityContentRow
  variant="non-user"
  avatarSrc="https://avatars.githubusercontent.com/u/94311139?s=200&v=4"
  name="Amazon"
  walletAddress="0x1234567890abcdef1234567890abcdef12345678"
  amount={0.321}
  totalFollowers={123}
  tags={[
    { label: 'keyboard', value: 34 },
    { label: 'ergonomic', value: 56 },
    { label: 'wireless', value: 12 },
    { label: 'gaming', value: 77 },
    { label: 'work', value: 11 },
    { label: 'home', value: 34 },
  ]}
>
  <Text variant="body" className="text-primary-foreground">
    Extra Content
  </Text>
</IdentityContentRow>
`} />

<Canvas of={Stories.NonUserVariant} sourceState="none" />
