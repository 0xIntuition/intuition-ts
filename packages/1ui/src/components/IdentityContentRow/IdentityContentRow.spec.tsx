import React from 'react'

import { render, screen } from '@testing-library/react'
import { describe, expect, it } from 'vitest'

import { IdentityContentRow } from './IdentityContentRow'

describe('IdentityContentRow', () => {
  it('should render with required props', () => {
    render(
      <IdentityContentRow
        variant="user"
        name="John Doe"
        walletAddress="0x1234567890abcdef1234567890abcdef12345678"
        avatarSrc="https://avatars.githubusercontent.com/u/94311139?s=200&v=4"
        amount="1.210 ETH"
        totalFollowers={305}
      />,
    )

    expect(screen.getByText('John Doe')).toBeDefined()
    expect(screen.getByText('1.210 ETH')).toBeDefined()
    expect(screen.getByText('305')).toBeDefined()
    expect(screen.getByText('0x1234...cdef')).toBeDefined()
  })

  // Test if the component renders with tags
  it('should render with tags', () => {
    const tags = [
      { label: 'keyboard', value: 34 },
      { label: 'ergonomic', value: 56 },
      { label: 'wireless', value: 12 },
      { label: 'gaming', value: 77 },
      { label: 'work', value: 11 },
      { label: 'home', value: 34 },
    ]

    render(
      <IdentityContentRow
        variant="user"
        name="John Doe"
        walletAddress="0x1234567890abcdef1234567890abcdef12345678"
        avatarSrc="https://avatars.githubusercontent.com/u/94311139?s=200&v=4"
        amount="1.210 ETH"
        totalFollowers={305}
        tags={tags}
      />,
    )

    expect(screen.getByText('keyboard')).toBeDefined()
    expect(screen.getByText('ergonomic')).toBeDefined()
    expect(screen.getByText('wireless')).toBeDefined()
    expect(screen.getByText('gaming')).toBeDefined()
    // Ensure only the first 4 tags are rendered
    expect(screen.queryByText('work')).not.toBeDefined()
    expect(screen.queryByText('home')).not.toBeDefined()
  })

  // Test if the component renders children
  it('should render children', () => {
    render(
      <IdentityContentRow
        variant="user"
        name="John Doe"
        walletAddress="0x1234567890abcdef1234567890abcdef12345678"
        avatarSrc="https://avatars.githubusercontent.com/u/94311139?s=200&v=4"
        amount="1.210 ETH"
        totalFollowers={305}
      >
        <p>Extra Content</p>
      </IdentityContentRow>,
    )

    expect(screen.getByText('Extra Content')).toBeDefined()
  })

  // Snapshot test
  it('should match snapshot', () => {
    const { asFragment } = render(
      <IdentityContentRow
        variant="user"
        name="John Doe"
        walletAddress="0x1234567890abcdef1234567890abcdef12345678"
        avatarSrc="https://avatars.githubusercontent.com/u/94311139?s=200&v=4"
        amount="1.210 ETH"
        totalFollowers={305}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
