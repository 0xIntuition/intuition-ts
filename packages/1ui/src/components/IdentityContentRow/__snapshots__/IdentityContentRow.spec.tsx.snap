// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IdentityContentRow > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full mb-4"
  >
    <div
      class="w-full flex justify-between items-center"
    >
      <div
        class="flex items-center"
      >
        <span
          class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full mr-4"
        >
          <span
            class="bg-muted flex h-full w-full items-center justify-center rounded-full"
          >
            Jo
          </span>
        </span>
        <div
          class="flex flex-col"
        >
          <div
            class="flex items-center"
          >
            <p
              class="text-primary text-lg font-normal mr-1"
            >
              John Doe
            </p>
            <p
              class="text-base font-normal text-secondary-foreground"
            >
              0x1234...5678
            </p>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col items-end"
      >
        <p
          class="text-primary text-lg font-normal"
        >
          1.210 ETH
        </p>
        <div
          class="flex gap-1 items-center"
        >
          <svg
            class="text-secondary-foreground h-4 w-4"
          >
            <use
              href="/src/components/Icon/Icon.sprites.svg#people"
            />
          </svg>
          <p
            class="text-base font-normal text-secondary-foreground"
          >
            305
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
